### PR TITLE
nixos/consul: add types

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change
#76184

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by cole-h — CHANGES_REQUESTED** [review]
Looks like you also added `services.bind` types in here. Mind separating the commits? I'm fine with them both being in this PR, as long as the commits themselves are separate. (Or you can make a separate PR, up to you.)

**Comment by Scriptkiddi** [comment]
> Looks like you also added `services.bind` types in here. Mind separating the commits? I'm fine with them both being in this PR, as long as the commits themselves are separate. (Or you can make a separate PR, up to you.)

I didnt want to add bind here there is a seperate pr for that thanks for letting me know

File: nixos/modules/services/networking/consul.nix [hunk 1]
```diff
@@ -99,6 +99,7 @@ in
 
       extraConfig = mkOption {
         default = { };
+        type = types.lines;
```
Comment (cole-h) [inline nixos/modules/services/networking/consul.nix:102]: I don't think this is the right type.

```suggestion
        type = types.attrsOf types.str;
```

File: nixos/modules/services/networking/consul.nix [hunk 2]
```diff
@@ -99,6 +99,7 @@ in
 
       extraConfig = mkOption {
         default = { };
+        type = types.attrsOf types.str;
```
Comment (cole-h) [inline nixos/modules/services/networking/consul.nix:102]: I tried running the consul NixOS test locally and it failed cuz not everything is a `types.str`.

```suggestion
        type = types.attrsOf types.anything;
```

**Review by cole-h — APPROVED** [review]
Diff LGTM, thanks!

**Pull request merged**